src/routes/RequestQueueRoute.js

```diff
@@ -90,17 +106,76 @@ class RequestQueueRoute extends React.Component {
       }),
       requests: PropTypes.shape({
         records: PropTypes.array,
+      }).isRequired,
+      configs: PropTypes.shape({
+        records: PropTypes.array.isRequired,
+        hasLoaded: PropTypes.bool.isRequired,
       }),
     }),
+    match: PropTypes.object.isRequired,
     mutator: PropTypes.shape({
-      reorder: PropTypes.object,
+      reorderInstanceQueue: PropTypes.object.isRequired,
+      reorderItemQueue: PropTypes.object.isRequired,
+      requests: PropTypes.object.isRequired,
     }),
     history: PropTypes.shape({
       push: PropTypes.func.isRequired,
       goBack: PropTypes.func.isRequired,
     }).isRequired,
   };
 
+  componentDidMount() {
+    const {
+      mutator: { requests },
+      resources: {
+        configs,
+      },
+      match: {
+        params,
+      },
+    } = this.props;
+
+    this.getRequests(configs, requests, params.id);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { configs: prevConfigs } = prevProps.resources;
+    const {
+      mutator: { requests },
+      resources: {
+        configs,
+      },
+      match: {
+        params,
+      },
+    } = this.props;
+
+    if ((prevConfigs.hasLoaded !== configs.hasLoaded && configs.hasLoaded) || prevConfigs.records[0]?.value !== configs.records[0]?.value) {
+      this.getRequests(configs, requests, params.id);
+    }
+  }
+
+  getRequests = (configs, requestsMutator, id) => {
```

Comment:
looks like we do not need to pass requestsMutator and config, we can get them from props

Reply:
Done